### ISSUE TITLE
Regression: Xamarin.Forms Button Image position broken on Android

### ISSUE THREAD

**Issue by napalmDaz** [xamarin/Xamarin.Forms#4354]
### Description

When setting an image, text and ContentLayout on a button, the image is not positioned correctly on Android when using Xamarin Forms 3.3. This worked as expected in XF 3.1. iOS does not seem to be affected.

There are a couple problems here
1)  the code here https://github.com/xamarin/Xamarin.Forms/commit/3090843ce8f559f0c46fe08b68151721019b572d needs to be copied to the fast renderer
2) The code used to check if the padding has been set won't work because the Padding BP uses a defaultValueCreator and because it uses a defaultValueCreator then IsSet isn't reliable to check because ClearValue doesn't cause IsSet to return to false

### Steps to Reproduce

```xml
<?xml version="1.0" encoding="utf-8"?>
<ContentPage xmlns="http://xamarin.com/schemas/2014/forms" xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml" xmlns:local="clr-namespace:Buttons" x:Class="Buttons.MainPage">
<Grid x:Name="bottomLayout" ColumnSpacing="0" Margin="0" Padding="0">
    <Grid.ColumnDefinitions>
        <ColumnDefinition Width="*"/>
            <ColumnDefinition Width="74"/>
            <ColumnDefinition Width="74"/>
        </Grid.ColumnDefinitions>
        <StackLayout Orientation="Horizontal" x:Name="RoomSelectionOptions" Grid.Column="0" >
			<!-- Things -->
        </StackLayout>
        <Button Grid.Column="1" 
            x:Name="ControlsButton" 
            HorizontalOptions="EndAndExpand" 
            BackgroundColor="Transparent" 
            ContentLayout="Top,0" 
            FontSize="Micro" 
            Image="ic_brightness50" 
            TextColor="{StaticResource PrimaryDark}" 
            Text="Control" 
            Clicked="ControlsCommandButton_Clicked" />
        <Button Grid.Column="2" 
            x:Name="PowerCommandButton" 
            HorizontalOptions="Start" 
            BackgroundColor="Transparent" 
            ContentLayout="Top,0" 
            FontSize="Micro" 
            Image="ic_powerswitch" 
            TextColor="{StaticResource PrimaryDark}" 
            Text="Power" 
            Clicked="PowerCommandButton_Clicked" />
    </Grid>
</ContentPage>
```

### Expected Behavior

Button image is centered as in previous versions (and also iOS) and relationship specified in ContentLayout is respected.

### Actual Behavior

On Android, the button image is offset (no margin / padding) and not displayed in the expected location towards the center of the button.

### Basic Information

- Version with issue: 3.3.0.967583
- Last known good version: 3.1
- IDE: VS 2017 Enterprise
- Platform Target Frameworks: 
  - Android: 8.1 Oreo

### Screenshots

https://imgur.com/f3PSDDA

**Comment by PureWeen** [comment]
@napalmDaz  are you using Fast Renderers??


It looks like this PR needs to be copied to the Fast Renderer

https://github.com/xamarin/Xamarin.Forms/commit/3090843ce8f559f0c46fe08b68151721019b572d

**Comment by napalmDaz** [comment]
@PureWeen Unless fast renderers are enabled by default in XF3.3, then I don't think so as I'm not opting in via the flag in MainActivity. 

Here is a view hierarchy dump for confirmation:
https://imgur.com/a/VaLMW1p

**Comment by PureWeen** [comment]
@napalmDaz   Are you updated to AppCompat?  The non AppCompat button renderer also doesn't have the code in place to preserve the padding so maybe that's it?

**Comment by PureWeen** [comment]
Alright I'm guessing it's this line
https://github.com/xamarin/Xamarin.Forms/pull/2426/files#diff-08a5663cd61b123664c0ac0fd19c3255R62

Either way I've seen a few things that need to be fixed so we'll get this addressed.

**Comment by mattleibow** [comment]
I have fixed this in https://github.com/xamarin/Xamarin.Forms/pull/4967

**Comment by SaskiaL** [comment]
I just upgraded Xamarin.Forms to version 3.6.0.220655 and the ImagePosition of the buttons was no longer as it should be.

See attached picture with the buttons as they should be, and how they are in this latest version.

I have downgraded Xamarin.Forms to version 3.1.0.697729 and the buttons are again as they should be.

Kind regards, Saskia

![ImagePositionBroken](https://user-images.githubusercontent.com/20900758/54437759-9fb83880-4735-11e9-92dc-99ea3a2a84e9.png)
